Chitinid

ElusiveCoin opened the pull request:
<!-- If you are new to the Delta-V repository, please read the [Contributing Guidelines](https://github.com/DeltaV-Station/Delta-v/blob/master/CONTRIBUTING.md) -->

## About the PR
<!-- What did you change? -->
My first dip into content creation for SS14, i have mane the Chitinid, a species of ant people created as worker drones who are now their own people. Feedback welcomed as it is my first time.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
I wanted to try something new and take my own crack at content creation for the server, i had gotten the idea of worker ants/ drones of bug species and thought what if i threw a bit of firefly in there and some radiation and i just let the idea flow.
For balance sake i tried to stay light with things since it's my first time making content and aimed for something that would be good at general labor and rescuing but lacked in the combat department in the conventional ways making them just a different spice to play as.

## Technical details
<!-- Summary of code changes for easier review. -->
Some new code for a radiation purging system that is somewhat similar to the felinid's hairball, as well as minor adjustments to the injector code to allow the species to be tagged as unable to be injected.

## Media
<!-- Attach media if the PR makes in game changes (clothing, items, features, etc).
Small fixes/refactors are exempt. -->
![image](https://github.com/user-attachments/assets/ec0dbfa3-f6e1-4683-b61d-41974ac4e147)
![image](https://github.com/user-attachments/assets/df218703-c99b-4866-9b52-0debe6874bc0)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [ X] I have tested all added content and changes.
- [X ] I have added media to this PR or it does not require an in game showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->
No breaking changes.

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelogs must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl:
- add: Added the Chitinid species, fun little glowy ant guys!

ElusiveCoin committed:
Start of the Chitinid Race

My first work for ss14 and a new Species to play, worker drone ants!

ElusiveCoin committed:
Removed a uneeded bit

ElusiveCoin committed:
Removing the displacement section because it still is being weird

Displacement map not working right but not really needed as well so just gonna do without it

deltanedas commented on .run/Content Server+Client.run.xml:
untroll

ElusiveCoin replied:
I think i fixed it?

deltanedas commented on Resources/Locale/en-US/_DV/chat/managers/chat_manager.ftl:
```diff
@@ -21,3 +21,9 @@ chat-speech-verb-rodentia-1 = squeaks
+chat-speech-verb-chitinid-1 = clicks
+chat-speech-verb-chitinid-2 = chitters
+chat-speech-verb-chitinid-3 = hisses
+chat-speech-verb-chitinid-4 = Buzzes
```
buzzes, lowercase

ElusiveCoin replied:
Corrected, will Commit with the rest

deltanedas commented on Resources/Locale/en-US/_DV/species/species.ftl:
```diff
@@ -3,3 +3,4 @@
 species-name-vulpkanin = Vulpkanin
 species-name-harpy = Harpy
 species-name-rodentia = Rodentia
+species-name-chitinid = Chitinid
```
trailing newline

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Body/Organs/chitinid.yml:
```diff
@@ -0,0 +1,33 @@
+    metabolizerTypes: [Human]
+    groups:
+    - id: Food
+    - id: Drink
```
trailing newline

ElusiveCoin replied:
fixed

deltanedas commented on Resources/Prototypes/_DV/Body/Parts/chitinid.yml:
```diff
@@ -0,0 +1,120 @@
+  parent: [PartChitinid, BaseTorso]
+  components:
+  - type: Sprite
+    sprite: _DV/Mobs/Species/Chitinid/parts.rsi
```
put this path on PartChitinid and remove it from all limbs

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Body/Parts/chitinid.yml:
```diff
@@ -0,0 +1,120 @@
+# TODO: Add descriptions (many)
+# TODO BODY: Part damage
+- type: entity
+  id: PartChitinid
```
rename to PartChitinidBase

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Damage/modifier_sets.yml:
```diff
@@ -29,3 +29,13 @@
+    Slash: 0.9
+    Cold: 1.1
+    Poison: 0.8
+    Radiation: 0.2
```
trailing newline

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Damage/modifier_sets.yml:
```diff
@@ -29,3 +29,13 @@
+    Piercing: 1.25
+    Slash: 0.9
+    Cold: 1.1
+    Poison: 0.8
```
this wont do anything as reagents dont use damage modifiers and no weapons do poison damage

ElusiveCoin replied:
Fixed, Whoops didn't mean to leave that in there!

deltanedas commented on Resources/Prototypes/_DV/Datasets/Names/chitinid_first_female.yml:
```diff
@@ -0,0 +1,41 @@
+- type: dataset
+  id: names_chitinid_first_female
```
use PascalCase for ids also sort these alphabetically

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Datasets/Names/chitinid_first_male.yml:
```diff
@@ -0,0 +1,42 @@
+- type: dataset
+  id: names_chitinid_first_male
```
above

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Datasets/Names/chitinid_last.yml:
```diff
@@ -0,0 +1,31 @@
+- type: dataset
+  id: names_chitinid_last
```
above

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Entities/Mobs/Species/chitinid.yml:
```diff
@@ -0,0 +1,167 @@
+    color: "#ff7a00"
+    autoRot: true
+
+  - type: Carriable # Carrying system from nyanotrasen.
```
dont need this comment

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Entities/Mobs/Species/chitinid.yml:
```diff
@@ -0,0 +1,167 @@
+    damageModifierSet: Chitinid
+
+  - type: ZombieAccentOverride
+    accent: zombie
```
this is for moth and stuff that have a special accent instead of the default, remove

ElusiveCoin replied:
Thought so, this is why ya don't do stuff at 2AM, Fixed

deltanedas commented on Resources/Prototypes/_DV/Entities/Mobs/Species/chitinid.yml:
```diff
@@ -0,0 +1,167 @@
+    modifiers:
+      coefficients:
+        Blunt: 1.1
+        Slash: 1.1
```
this makes them do more melee damage, not less

it also applies prying modifier and extreme gun spread so can't be used without oni refactoring into separate components, see #2708

ElusiveCoin replied:
Ah So thats what that PR was about, ya non of that is what was in mind so removing that, Fixed

deltanedas commented on Resources/Prototypes/_DV/Voice/speech_emote_sounds.yml:
```diff
@@ -242,3 +242,37 @@
+    Gasp:
+      collection: MaleGasp
+    DefaultDeathgasp:
+      collection: MothDeathGasp
```
trailing newline

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Voice/speech_emote_sounds.yml:
```diff
@@ -242,3 +242,37 @@
+    Yawn:
+      collection: MaleYawn
+    Hiss:
+      path: /Audio/Animals/snake_hiss.ogg # DeltaV
```
dont need the comment

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/Voice/speech_sounds.yml:
```diff
@@ -15,3 +15,12 @@
+  askSound:
+    path: /Audio/Voice/Talk/speak_1_ask.ogg
+  exclaimSound:
+    path: /Audio/Voice/Talk/speak_1_exclaim.ogg
```
trailing newline

ElusiveCoin replied:
fixed

deltanedas commented on Resources/Prototypes/_DV/Voice/speech_verbs.yml:
```diff
@@ -33,3 +33,12 @@
+  - chat-speech-verb-chitinid-1
+  - chat-speech-verb-chitinid-2
+  - chat-speech-verb-chitinid-3
+  - chat-speech-verb-chitinid-4
```
trailing newline

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Prototypes/_DV/typing_indicator.yml:
```diff
@@ -10,3 +10,8 @@
+  id: Chitinid
+  spritePath: /Textures/_DV/Effects/speech.rsi
+  typingState: chitinid0
+  offset: -0.2, 0.1 # 0625
```
trailing newline

ElusiveCoin replied:
fixed

deltanedas commented on Resources/ServerInfo/Guidebook/Mobs/_DV/Chitinid.xml:
```diff
@@ -0,0 +1,26 @@
+  ## Drawbacks
+  - Built for work rather then combat their hard shells are weaker to Blunt and piercing damage, they take 25% more damage to both.
+  - Thanks to their overactive systems they get hungry 33% faster.
+  - The cold dose not agree with their biology and makes their movement slugish, the cold also harms them more then others.
```
dose -> does
slugish -> sluggish
then -> than

ElusiveCoin replied:
Thank you, fixed

deltanedas commented on Resources/ServerInfo/Guidebook/Mobs/_DV/Chitinid.xml:
```diff
@@ -0,0 +1,26 @@
+    <GuideEntityEmbed Entity="MobChitinid" Caption=""/>
+  </Box>
+
+  A hard working worker drone species, the Chitinid are strong dilligent workers, thanks to their homeworld's enviroment they have grown an acute resistant to radiation but not without it's side effects.
```
An industrious worker drone species
dilligent -> diligent
, thanks -> . Thanks
, after environment
resistant -> resistance
, after radiation
it's -> its

ElusiveCoin replied:
Fixed much cleaner

deltanedas commented on Resources/Textures/_DV/Effects/speech.rsi/meta.json:
```diff
@@ -40,6 +40,23 @@
         {
             "name": "rodentia2"
+        },
+		{
```
dont mix tabs and spaces

ElusiveCoin replied:
Fixed

deltanedas commented on Resources/Textures/_DV/Mobs/Customization/Chitinid/chitinid_antennas.rsi/meta.json:
4 space indent for meta.json files

ElusiveCoin replied:
Fixed it

deltanedas replied:
nope its still 2 space

deltanedas commented on Resources/Textures/_DV/Mobs/Customization/Chitinid/chitinid_parts.rsi/meta.json:
.

deltanedas commented on Resources/Textures/_DV/Mobs/Customization/Chitinid/chitinid_parts.rsi/meta.json:
```diff
@@ -0,0 +1,143 @@
+      "directions": 4
+    }
+  ]
+}
```
trailing newline

ElusiveCoin replied:
fixed

deltanedas commented on Resources/Textures/_DV/Mobs/Customization/Chitinid/chitinid_wings.rsi/meta.json:
.

deltanedas commented on Resources/Textures/_DV/Mobs/Species/Chitinid/parts.rsi/meta.json:
.

deltanedas commented on Resources/Textures/_DV/Mobs/Species/Chitinid/parts.rsi/meta.json:
```diff
@@ -0,0 +1,66 @@
+      "directions": 4
+    }
+  ]
+}
```
trailing newline

ElusiveCoin replied:
fixed

rosieposieeee committed:
Nukie Steamroll Preventative (#2658)

* Update uplink_catalog.yml

* Update uplink_catalog.yml

* Update uplink_catalog.yml

* Update uplink_catalog.yml

DeltaV-Bot committed:
Automatic changelog update

ElusiveCoin committed:
most asked for changes

ElusiveCoin committed:
(hopefully) Fixed Linter Error

ElusiveCoin committed:
Merge branch 'master' into Chitinid

dvir001 commented:
You don't need to say fixed for every single fix, you are spamming the web hook.

![image](https://github.com/user-attachments/assets/2b486a1c-d68e-4482-9371-5385b4a5a2c5)

Also your change log is broken, use the format as per baseline or its not going to show in game.

deltanedas commented on Resources/Audio/_DV/Voice/Chitinid/attributions.yml:
```diff
@@ -0,0 +1,9 @@
+- files: ["moth_laugh.ogg, moth_chitter.ogg, moth_squeak.ogg"]
+  license: "CC-BY-SA-3.0"
+  copyright: "Taken from https://github.com/BeeStation/BeeStation-Hornet/commit/11ba3fa04105c93dd96a63ad4afaef4b20c02d0d"
+  source: "https://github.com/BeeStation/BeeStation-Hornet/blob/11ba3fa04105c93dd96a63ad4afaef4b20c02d0d/sound/emotes/"
```
trailing newline

deltanedas commented on Resources/Prototypes/_DV/Body/Organs/chitinid.yml:
```diff
@@ -0,0 +1,34 @@
+    groups:
+    - id: Food
+    - id: Drink
+    
```
\r\n stuff bad

deltanedas commented on Resources/Prototypes/_DV/Datasets/Names/chitinid_first_female.yml:
```diff
@@ -0,0 +1,41 @@
+- type: dataset
+  id: Names_Chitinid_First_Female
```
no _

deltanedas commented on Resources/Prototypes/_DV/Datasets/Names/chitinid_first_male.yml:
```diff
@@ -0,0 +1,42 @@
+- type: dataset
+  id: Names_Chitinid_First_Male
```
no _

deltanedas commented on Resources/Prototypes/_DV/Datasets/Names/chitinid_last.yml:
```diff
@@ -0,0 +1,31 @@
+- type: dataset
+  id: Names_Chitinid_Last
```
no _

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
More clean up edits

ElusiveCoin committed:
Merge branch 'Chitinid' of https://github.com/ElusiveCoin/Delta-v_LucyTime into Chitinid

Lyndomen commented:
Howdy! Thank you for presenting this, we are taking it under Direction review. As this is quite large, I estimate it will take three days.

ElusiveCoin committed:
4 space indents!!

Lyndomen commented:
Some of the initial feedback we have is that this is AWESOME, and super lovely sprites. We'd also like to request a bit of the default-names to be changed, we aren't picking up a suitable alien naming pattern, and would appreciate a more "out there" approach. 

Additionally. I personally love the radiation theming that this species has. I'd be curious to hear on if you have any ideas for how else this theme could be mechanically reinforced further. One of the ideas floated was to give them a radioactive purging mechanism, similar to the felinid's hairball- but resulting in a radioactive rock of some ilk.

ElusiveCoin commented:
> Some of the initial feedback we have is that this is AWESOME, and super lovely sprites. We'd also like to request a bit of the default-names to be changed, we aren't picking up a suitable alien naming pattern, and would appreciate a more "out there" approach.
> 
> Additionally. I personally love the radiation theming that this species has. I'd be curious to hear on if you have any ideas for how else this theme could be mechanically reinforced further. One of the ideas floated was to give them a radioactive purging mechanism, similar to the felinid's hairball- but resulting in a radioactive rock of some ilk.

Im happy to hear that!! Normally im not the best with names but with that kinda direction point i think i can totally make those names more alien like in nature and pattern i already have some idea brewing!

As for more ideas to mechanically play on the radiation theme i had a few but not many i didn't wanna try and go too far with much but i played with the idea of their unarmed attack/bite doing a bit of radiation damage, as well as the idea of them gaining som kind of buff at specific levels of radiation damage. I also played with the silly idea of lettingnthem consume uranium at the cost of some level of harm, but i thought it was a tad silly, however i love the purging idea making a radioactive rock!! Any more input or ideas would be welcomed i think im 100% gonna work on that purging idea it just sounds like a cool idea plus, funny radioactive rock.

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Direction desired name changes + More radiation mechanics

Alienized the names of the little Ants and swapped their naming scheme. Added a radiation healing and purge mechanic to play into the radiation theme more.

ElusiveCoin commented:
- Changed Bio-light to radioactive Green to better fit the radiation theme.
- Swapped them to a first name only naming convention along with making more exotic alien names for them and to fit into the drone species aesthetic (Hopefully its a nice pattern).
- Added a radiation purge mechanic, Their bodies naturally heal from radiation damage over time up to 30 points of radiation damage, once this cap is hit they will resume taking radiation damage until they cough up the crystalized radiation from their system, once the stone is purged they will resume healing it again but can not cough up another stone for 5 minutes.
- Added Chitzite, a small lightly radioactive stone, this is the result of them purging their system of radiation.
- Chitzite can be used as a weak weapon dealing small rad damage, if ya manage to form one its a good early alternative for Epi when they need a uranium spear. 

What do ya think of this interpretation of your guys idea for the radiation purge? dose that feed into the theme more mechanically in a nice way? @Lyndomen

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

Lyndomen commented:
How do we feel about their exoskeleton disabling injection via syringe?

ElusiveCoin commented:
> How do we feel about their exoskeleton disabling injection via syringe?

Hmmm I like the idea, on one hand tho their inherent weakness to piercing makes me struggle to see how to logically explain it but i think i can see a logical reason as to why (AKA weaker smaller needles can't get through as well as a spear or bullet)

But on the other hand the idea that their tougher to heal because of the tough shell and require more then just a quick injection to fix is nice, means pills or feeding meds in liquid form are used a bit more often which is just a nice little change.

So id say personally im down for either way, just what are your thoughts on what i said?

ElusiveCoin committed:
Merge branch 'master' into Chitinid

Lyndomen commented on Resources/Prototypes/_DV/Damage/modifier_sets.yml:
```diff
@@ -29,3 +29,12 @@
+- type: damageModifierSet
+  id: Chitinid
+  coefficients:
+    Blunt: 1.25
```
Blunt: 1.15 please!

Lyndomen replied:
Once that is all set, Direction approved! We're super happy to see these critters at work!!

ElusiveCoin replied:
Awesome! Changes are made and your suggestion of making them un-injectable via normal syringes has been coded and added and tested! once i get confirmation that the way the addition was coded is sound I'll commit it!!

ElusiveCoin replied:
Actually for ease of it being read here i will commit the changes, and either need to fix it first or get the funny thumbs up that it's good to go!

ElusiveCoin committed:
Hopefully final changes!

Adjustment to their resistances to make them less weak to blunt

And the negative of being unable to be injected by normal syringes (hypo and pill gaming)

ElusiveCoin committed:
Forgot to add access to the new undergarments

FaintSpeaker commented:
@Lyndomen thoughts on whether there should be a message saying "Exoskeleton too thick!" Or something similar to alert doctors/newbies what is happening instead of silently failing to inject?

Lyndomen commented:
> "Exoskeleton too thick!"

Yes, there absolutely should be a mechanism for notifying injectors that there is an issue.

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Final Final change (hopefully)

Added a little text pop up for failing to inject them

ElusiveCoin commented:
> > "Exoskeleton too thick!"
> 
> Yes, there absolutely should be a mechanism for notifying injectors that there is an issue.

Alright simple and added, now when ya try and inject them it tells ya exactly that! @Lyndomen that good?

Lyndomen reviewed (dismissed):
Direction approved!!

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

BarryNorfolk commented on Resources/Prototypes/Entities/Mobs/Customization/Markings/gauze.yml:
```diff
@@ -226,7 +226,7 @@
   bodyPart: Head
   markingCategory: Overlay
-  speciesRestriction: [Dwarf, Human, Reptilian, Arachnid, Moth]
+  speciesRestriction: [Dwarf, Human, Reptilian, Arachnid, Moth, Chitinid]
```
I believe lines like these (That sit outside of a _DV folder) need a marking like #DeltaV. Check the https://github.com/DeltaV-Station/Delta-v/blob/master/CONTRIBUTING.md guidelines and I think it explains it.

deltanedas replied:
yup

deltanedas commented on Content.Server/Chemistry/EntitySystems/InjectorSystem.cs:
```diff
@@ -108,6 +109,12 @@ private void OnInjectorAfterInteract(Entity<InjectorComponent> entity, ref After
+        {
+            Popup.PopupEntity(Loc.GetString("injector-component-deny-user"), target, user);
+            return;
+        }
```
needs deltav comments

deltanedas commented on Content.Server/Chemistry/EntitySystems/InjectorSystem.cs:
```diff
@@ -253,6 +260,9 @@ private bool TryInjectIntoBloodstream(Entity<InjectorComponent> injector, Entity
         Entity<SolutionComponent> targetSolution, EntityUid user, bool asRefill)
     {
+        if (HasComp<BlockInjectionComponent>(targetEntity))
+            return false;
```
above

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+
+namespace Content.Server.Abilities.Chitinid;
+
+
```
1 blank line not 2

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+public sealed partial class ChitinidComponent : Component
+{
+    [DataField("ChitzitePrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string ChitzitePrototype = "Chitzite";
```
replace with
```cs
[DataField]
public EntProtoId ChitzitePrototype = "Chitzite";
```

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+
+    [DataField("ChitziteActionId",
+        customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string? ChitziteActionId = "ActionChitzite";
```
above

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+        customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string? ChitziteActionId = "ActionChitzite";
+
+    [DataField("ChitziteAction")]
```
just [DataField]

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+    public EntityUid? ChitziteAction;
+
+    [DataField]
+    public float AmountAbsorbed { get; set; } = 0f;
```
remove { get; set; } and change float to FixedPoint2

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+    public float AmountAbsorbed { get; set; } = 0f;
+
+    [DataField]
+    public float MaximumAbsorbed = 30f;
```
FixedPoint2

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+    public float MaximumAbsorbed = 30f;
+
+    [DataField]
+    public TimeSpan UpdateInterval { get; set; } = TimeSpan.FromSeconds(1);
```
remove { get; set; }

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidComponent.cs:
```diff
@@ -0,0 +1,32 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
```
remove the .Prototype line

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+
+public sealed partial class ChitinidSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
```
rename to _actions

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
```
rename to _proto

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
```
rename to _popupSystem and use SharedPopupSystem

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
```
_inventory

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
```
_timing

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+        base.Initialize();
+        SubscribeLocalEvent<ChitinidComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<ChitinidComponent,ChitziteActionEvent>(OnChitzite);
+        SubscribeLocalEvent<ChitinidComponent,MapInitEvent>(OnMapInit);
```
space after ,

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ChitinidComponent, ComponentInit>(OnInit);
```
remove

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+        SubscribeLocalEvent<ChitinidComponent,MapInitEvent>(OnMapInit);
+    }
+
+    private Queue<EntityUid> RemQueue = new();
```
use RemCompDeferred and remove this

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+        {
+            RemComp<CoughingUpChitziteComponent>(bug);
+        }
+        RemQueue.Clear();
```
remove

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+            if (chitinid.AmountAbsorbed >= chitinid.MaximumAbsorbed && !_mobState.IsDead(uid))
+                continue;
+
+            var damage = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Radiation"), -0.5);
```
store this on the component

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+
+            var damage = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Radiation"), -0.5);
+            var delta = _damageable.TryChangeDamage(uid, damage);
+            if (delta != null)
```
`if (_damageable.TryChangeDamage(uid, damage, damageable: damageable) is {} delta)`

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+
+            chitinid.NextUpdate += chitinid.UpdateInterval;
+
+            if (chitinid.AmountAbsorbed >= chitinid.MaximumAbsorbed && !_mobState.IsDead(uid))
```
replace `&& !` with `||`, this currently makes dead chitinid regen radiation forever

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+            var delta = _damageable.TryChangeDamage(uid, damage);
+            if (delta != null)
+            {
+                chitinid.AmountAbsorbed += -(delta.GetTotal().Float());
```
dont need () around this

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+            }
+        }
+
+        foreach (var (chitziteComponent, chitinidComponent) in EntityQuery<CoughingUpChitziteComponent, ChitinidComponent>())
```
use EntityQueryEnumerator, replace Accumulator copy pasted from felenid with timespan

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+using Content.Server.Abilities.Felinid;
```
:cat:

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+        if (component.ChitziteAction != null)
+            return;
+
+        _actionsSystem.AddAction(uid, ref component.ChitziteAction, component.ChitziteActionId);
```
move this to MapInit handler

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+        _actionsSystem.AddAction(uid, ref component.ChitziteAction, component.ChitziteActionId);
+    }
+
+    private void OnChitzite(EntityUid uid, ChitinidComponent component, ChitziteActionEvent args)
```
`Entity<ChitinidComponent> ent, ref ChitziteActionEvent args` and changes stuff to use ent.Comp instead of component accordingly

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+    private void OnChitzite(EntityUid uid, ChitinidComponent component, ChitziteActionEvent args)
+    {
+        if (_inventorySystem.TryGetSlotEntity(uid, "mask", out var maskUid) &&
+            EntityManager.TryGetComponent<IngestionBlockerComponent>(maskUid, out var blocker) &&
```
TryComp

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+
+    private void SpawnChitzite(EntityUid uid, ChitinidComponent component)
+    {
+        var chitzite = EntityManager.SpawnEntity(component.ChitzitePrototype, Transform(uid).Coordinates);
```
Spawn instead of EntityManager.SpawnEntity

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs:
```diff
@@ -0,0 +1,123 @@
+        args.Handled = true;
+    }
+
+    private void SpawnChitzite(EntityUid uid, ChitinidComponent component)
```
take `Entity<ChitinidComponent> ent`

deltanedas commented on Content.Server/_DV/Abilities/Chitinid/CoughingUpChitziteComponent.cs:
```diff
@@ -0,0 +1,11 @@
+    [DataField("accumulator")]
+    public float Accumulator = 0f;
+
+    [DataField("coughUpTime")]
```
just [DataField] and replace accumulator with an AutoPausedField TimeSpan NextCough using TimeOffsetSerializer

deltanedas reviewed (changes requested):
copy pasting nyano shitcode bad

ElusiveCoin committed:
Merge remote-tracking branch 'upstream/master' into Chitinid

ElusiveCoin commented:
@deltanedas All requested changes made and commited! please re review when able.

ElusiveCoin committed:
Code Tweeks

ElusiveCoin commented:
Oops forgot to hit the commit button, NOW the changes are in.

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin committed:
Merge branch 'master' into Chitinid

Avalon-Proto commented:
A quick question, do Hypos, or at least the Gorlex and Hypo pen. Work on them? If not it's gonna be hellish if you roll nukie as an ant

ElusiveCoin commented:
> A quick question, do Hypos, or at least the Gorlex and Hypo pen. Work on them? If not it's gonna be hellish if you roll nukie as an ant

Oh ya, hypos still work on them, its just normal hand syringes that don't.

ElusiveCoin committed:
Merge branch 'master' into Chitinid

ElusiveCoin commented:
@deltanedas Ah i also forgot to state, the changes were also tested a few times to insure no issues popped up and all was clear so hopefully that means they were done right.

deltanedas commented on Resources/Prototypes/_DV/Actions/types.yml:
```diff
@@ -48,3 +48,16 @@
+- type: entity
+  id: ActionChitzite
+  name: action-name-chitzite
+  description: action-description-chitzite
```
doesnt need to use locale, downstreams can translate with entity locale overrides

deltanedas commented on Resources/Prototypes/_DV/Species/chitinid.yml:
dont need deltav comments on underwear stuff here

ElusiveCoin committed:
Quick Changes

Removed some uneeded comments, removed uneeded Local info

ElusiveCoin commented:
@deltanedas Quick changes have been made to correct them.

deltanedas commented:
amazing first pr, im gonna combine the chitzite system with felenid refactoring later

ElusiveCoin commented:
> amazing first pr, im gonna combine the chitzite system with felenid refactoring later

Thank you! that means a lot!

Avalon-Proto commented:
Did the publish not run this morning or did the CL break?

sleepyyapril commented:
Runs every morning at the same time. It'll be in tomorrow.

BlitzTheSquishy commented:
Made faster than feroxi :bluntsive:

ElusiveCoin commented:
Author note: for those curious if you wish to port them to a server with an MIT license you have my full express permission to do so
